lib/ecto/migration/runner.ex

```diff
@@ -263,10 +263,27 @@ defmodule Ecto.Migration.Runner do
   defp log_and_execute_ddl(repo, %{level: level, sql: sql}, command) do
     log(level, command(command))
     meta = Ecto.Adapter.lookup_meta(repo)
-    {:ok, _} = repo.__adapter__.execute_ddl(meta, command, timeout: :infinity, log: sql)
+    {:ok, notices} = repo.__adapter__.execute_ddl(meta, command, timeout: :infinity, log: sql)
+    log_notices(level, notices)
     :ok
   end
 
+  defp log_notices(_level, []), do: :ok
+  defp log_notices(level, [message | notices]) do
+    # TODO: what would we want to log??
+    %{
+      code: _code,
+      file: _file,
+      line: _line,
+      message: message,
+      routine: _routine,
+      severity: _severity
+    } = message
```

Comment:
I don't think we should access this, as this internal representation will change per adapter. My suggestion is for us to add a new callback to `Ecto.Adapters.SQL.Connection` called `ddl_logs` that receives each `Postgrex.Result` returned in `execute_ddl` and it must return a list in this shape: `{level, msg, metadata}`. Then on this side, we should call `Logger.log(level, msg, metadata)`.

Reply:
If we do in this direction, we should probably change [this](https://github.com/elixir-ecto/ecto/blob/master/lib/ecto/adapter/migration.ex#L57) so the return result is no longer `list()` but rather something like `[{Logger.level, Logger.message, Logger.metadata}]`.